Generate coverage report of glass repo

connoraird opened the pull request:
Generate a coverage report of the glass repo from the benchmarks in this repo

- Add glass as a submodule to allow displaying src files in coveralls

Saransh-cpp committed:
gh-352: fix latex rendering (#357)

Single backslash is for latex commands, double backslash is for newline.

Closes: #352

paddyroddy committed:
gh-280: add `mypy` to codebase (#359)

Adding `mypy` to a pre-existing codebase is never easy. I initially
attempted this in #308, but have since split this up into separate
issues:
- [x] #347
- [x] #356
- [ ] #358

In this PR, `mypy` is added but in order for CI to pass, the `# type:
ignore[]` syntax is used throughout. I didn't want to tackle these here
too (see #308) as it gets quite messy.

One thing I have done (following #356) is change every empty
`npt.NDArray` to `npt.NDArray[typing.Any]` (see #330), as this actually
results in fewer errors than leaving them all blank. Ideally, we'd like
to fill in as many of the `typing.Any` as possible - they're a bit
useless by themselves. However, that is not the priority for now. Plus,
I expect typing to change when #67 is tackled.

Saransh-cpp committed:
gh-248: a few fixes for ignored and recommended ruff rules (#360)

Fixed:
- missing-trailing-comma
- commented-out-code
- single-line-implicit-string-concatenation
- numpy-legacy-random
- pytest-raises-too-broad

Refs: #248

Saransh-cpp committed:
gh-363: fix flakiness of test_ellipticity_gaussian (#364)

Increase the tolerance for the test to pass always.

Closes: #363

Saransh-cpp committed:
gh-175: add CITATION.cff and bib entry in README (#325)

Adds a `.cff` file and moves the `bib` entry to README.

Closes: #175
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>
Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>

Saransh-cpp committed:
gh-131: rename `gaussian_gls` to `discretized_cls` (#345)

This popped up while I was looking to add tests for `fields.py`.

- rename `gaussian_gls` to `discretized_cls`
- update docstring and internal references
- don't use `gaussian_gls` in `lognormal_gls`

Closes: #131
Changed: `gaussian_gls` has been renamed to `discretized_cls`
Changed: `lognormal_gls` no longer uses `gaussian_gls` internally and
the arguments passed to `gaussian_gls` have been removed
Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>

mwiet committed:
NLA model for intrinsic alignments (#21)

The ia_nla function computes the effective convergence from intrinsic
alignments, calculated using the Non-linear Aligments Model (NLA). The
convergence from intrinsic aligments is follows Catelan et al. (2000),
Hirata & Seljak (2004), Bridle & King (2007).

Added: The new `kappa_ia_nla()` function computes an effective
convergence due to Intrinsic Alignments using the NLA model.
Reviewed-by: Nicolas Tessore
Reviewed-by: Arthur Loureiro <arthur.loureiro@fysik.su.se>

---------

Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

Saransh-cpp committed:
gh-161: bump coverage for galaxies and shapes (#370)

The coverage went down when branch coverage was turned on. This PR
brings it back to 100% for these two files.

Refs: #161
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
Fix `Returns` in docstrings (I think entered in #357) (#373)

Saransh-cpp committed:
gh-363: use pytest-rerunfailures for flaky test_ellipticity_gaussian (#371)

Rerun the flaky test if it fails. See
https://github.com/glass-dev/glass/pull/364#issuecomment-2414422872

Closes: #363

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#378)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Saransh Chopra <saransh0701@gmail.com>

paddyroddy committed:
gh-379: fix warnings when running the notebooks (#380)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#387)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

paddyroddy committed:
gh-362: review the `ruff-format` recommended rules (#393)

paddyroddy committed:
gh-354: tidy Actions, name steps, cache dependencies (#388)

Saransh-cpp committed:
gh-342: explicit libbz2-dev installation is not required now (#396)

paddyroddy committed:
gh-358: add static types support (#368)

paddyroddy committed:
gh-248: fix all remaining possible ruff `# noqa` statements (#394)

paddyroddy committed:
gh-385: set random seeds in all glass functions used in notebooks (#389)

Saransh-cpp committed:
gh-365: upload HTML rendering of notebooks as artifact (#367)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

Saransh-cpp committed:
gh-336: support Python 3.13 (#337)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-399: hide `kappa_ia_nla` (#400)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#412)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

Saransh-cpp committed:
gh-272: add tests for glass.points (#372)

paddyroddy committed:
gh-188: add docstrings to all functions and tidy docs (#381)

Co-authored-by: Saransh Chopra <saransh0701@gmail.com>
Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>

paddyroddy committed:
gh-419: fix imports (#420)

paddyroddy committed:
gh-286: replace `stable` in docs with tag for `sdist` (#416)

Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>

paddyroddy committed:
gh-418: add variable names for `Yields` vars (#428)

ntessore committed:
gh-390: release 2024.2 (#426)

This is the release PR for v2024.2.

Closes: #390

ntessore committed:
gh-430: fix for release action (#429)

paddyroddy committed:
gh-163: drop `Python 3.9` & `NumPy < 2.1.1` support (#391)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

paddyroddy committed:
gh-431: switch from `license.file` to `license.text` (#434)

paddyroddy committed:
Fix linting (#435)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#437)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

paddyroddy committed:
gh-384: rename all internal functions to `trapezoid` over `trapz` (#392)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

paddyroddy committed:
gh-443: `if` should be `elif` in `fixed_zbins` (#444)

paddyroddy committed:
gh-445: place all public function in `__all__` (#446)

paddyroddy committed:
gh-441: import directly from `glass` in tests (#442)

paddyroddy committed:
gh-447: make sure `dz` gives the same results as `nbins` (#448)

dependabot[bot] committed:
DEV: Bump thehanimo/pr-title-checker from 1.4.2 to 1.4.3 (#452)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

paddyroddy committed:
gh-450: move all `pytest.fixture` into `conftest.py` (#451)

Saransh-cpp committed:
gh-453: don't run PR title checker on dependabot PRs (#455)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#458)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

paddyroddy committed:
gh-456: call `redshift_grid` within `distance_grid` (#457)

Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>

paddyroddy committed:
gh-271: add tests for `glass.observations` (#436)

Saransh-cpp committed:
gh-269: add tests for glass.fields (#374)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

Saransh-cpp committed:
gh-274: add missing tests for `glass.user` (#422)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-368: add type overloading for `lensing.from_convergence` (#395)

paddyroddy committed:
gh-421: fix TestPyPI deployment (#427)

paddyroddy committed:
gh-398: consistent plotting for notebooks (#425)

Saransh-cpp committed:
gh-460: update notebooks inplace when running nox -s examples (#461)

Saransh-cpp committed:
gh-440: remove scipy as a test dependency (#462)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#466)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#467)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#477)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-472: fix Sphinx configuration (#473)

paddyroddy committed:
gh-469: @pre-commit.ci shouldn't break title checking (#474)

Saransh-cpp committed:
gh-470: instructions to install using conda (#481)

Saransh-cpp committed:
gh-464: simplify release infrastructure (#482)

Saransh-cpp committed:
gh-465: enable installing glass through git archives (#483)

paddyroddy committed:
Manual revert of #467 (#484)

paddyroddy committed:
gh-468: fix virtualenv in testing (#471)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#485)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

Saransh-cpp committed:
gh-439: infrastructure for testing array API compatibility (#459)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

Saransh-cpp committed:
gh-424: add a simple PR template (#463)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-275: add tests for `glass.lensing` (#454)

paddyroddy committed:
gh-273: add tests for `glass.shells` (#449)

Co-authored-by: Saransh Chopra <saransh0701@gmail.com>

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#489)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

Saransh-cpp committed:
gh-480: fix mypy failures (#490)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#498)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#500)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

Saransh-cpp committed:
gh-506: cache dependencies correctly (#505)

ntessore committed:
gh-493: add module for Gaussian random fields (#494)

Co-authored-by: Saransh Chopra <saransh0701@gmail.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

Saransh-cpp committed:
gh-499: rm D205, D401, EM101, EM102, TRY003 (#510)

Saransh-cpp committed:
gh-491: use try-except for __version__ (#511)

paddyroddy committed:
gh-503: move all typing related imports into `from` (#507)

paddyroddy committed:
gh-501: change imports to be directly from `glass` (#512)

Saransh-cpp committed:
gh-515: fix coveralls (#516)

paddyroddy committed:
gh-501: changes missed in #507 (#517)

paddyroddy committed:
gh-501: move custom types into `TYPE_CHECKING` (#520)

Saransh-cpp committed:
gh-480: fix union1d typing error (#521)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#522)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

ntessore committed:
gh-496: functions for legacy mode (#497)

Co-authored-by: Saransh Chopra <saransh0701@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

ntessore committed:
gh-525: fix docs (#526)

ntessore committed:
gh-513: make algorithm a public module (#514)

Co-authored-by: Saransh Chopra <saransh0701@gmail.com>

ntessore committed:
gh-508: create standalone bibliography page (#518)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>
Co-authored-by: Saransh Chopra <saransh0701@gmail.com>

Saransh-cpp committed:
gh-478: glass.core.array -> glass.arraytools (#524)

Saransh-cpp committed:
gh-509: `inv_triangle_number` -> `nfields_from_nspectra` (plus make it public) (#527)

Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>

ntessore committed:
gh-504: deprecate old generator functions (#523)

Co-authored-by: Saransh Chopra <saransh0701@gmail.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

Saransh-cpp committed:
gh-509: new inv_triangle_number popped up in another PR (#528)

paddyroddy committed:
Change the pre-commit updates to be monthly (#529)

paddyroddy committed:
gh-532: prepare for the `v2025.1` release (#531)

ntessore committed:
gh-530: more permissive typing of GLASS/HEALPix order functions (#534)

paddyroddy committed:
gh-538: suppress notebook rules rather than blanket turning off `ruff` (#539)

paddyroddy committed:
gh-540: add all contributors config (#541)

paddyroddy committed:
Rename .all-contributorsrc.json to .all-contributorsrc (#543)

allcontributors[bot] committed:
docs: add paddyroddy as a contributor for code (#544)

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

allcontributors[bot] committed:
docs: add Saransh-cpp as a contributor for code (#545)

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

allcontributors[bot] committed:
docs: add ucapbba as a contributor for code (#549)

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

allcontributors[bot] committed:
docs: add arthurmloureiro as a contributor for code (#550)

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

allcontributors[bot] committed:
docs: add mwiet as a contributor for code (#551)

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-547: comment out CHANGELOG entry by default (#548)

allcontributors[bot] committed:
docs: add joachimi as a contributor for ideas, and research (#556)

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Saransh Chopra <saransh0701@gmail.com>

allcontributors[bot] committed:
docs: add NiallJeffrey as a contributor for code, and ideas (#557)

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>
Co-authored-by: Saransh Chopra <saransh0701@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

allcontributors[bot] committed:
docs: add ARCHER2-HPC as a contributor for financial (#555)

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Saransh Chopra <saransh0701@gmail.com>

Saransh-cpp committed:
gh-546: ignore .all-contributorsrc from prettier (#560)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#564)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

paddyroddy committed:
gh-561: simplify @all-contributors config (#562)

paddyroddy committed:
gh-554: move contributors to a file and show in docs (#563)

paddyroddy committed:
gh-578: mutable argument should be empty list rather than `None` (#579)

Saransh-cpp committed:
gh-552: add docs on all-contributors (#565)

paddyroddy committed:
gh-569: enforce large file check with exception (#570)

paddyroddy committed:
gh-573: add keyword `rng` to functions in notebooks (#574)

Saransh-cpp committed:
gh-583: fix flakiness of `test_generate` (#584)

Saransh-cpp committed:
gh-575: bring back inv_triangle_number (#577)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-585: remove redunant `glass.ext.camb` import (#586)

Saransh-cpp committed:
gh-571: add a 20 min timeout to example runs (#582)

Saransh-cpp committed:
gh-580: examples should run in the CI if they are modified (#581)

Saransh-cpp committed:
gh-572: make `multalm` private (#576)

Saransh-cpp committed:
gh-487: reorder alms in _generate_grf to glass ordering (#533)

Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

Saransh-cpp committed:
gh-587: run examples if a PR is labeled 'run-examples' (#588)

paddyroddy committed:
gh-568: convert `_weight` funcs to classes for `_windows` funcs (#589)

Saransh-cpp committed:
gh-592: use `--doctest-plus-generate-diff=overwrite` (#593)

paddyroddy committed:
gh-495: allow `zeff` to be computed automatically (#590)

Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#598)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

Saransh-cpp committed:
gh-602: pin python version and numpy for pre-commit (#603)

paddyroddy committed:
gh-599: swap all `xp` vars to `xq` (and similar) (#600)

Saransh-cpp committed:
gh-601: add a utility function to fetch array namespace (#604)

ntessore committed:
gh-566: add example on galaxy redshift distributions (#567)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#615)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

paddyroddy committed:
gh-612: handle NoneType in `get_namespace` (#613)

paddyroddy committed:
gh-616: use `position_weights()` in galaxy redshift example (#617)

Saransh-cpp committed:
gh-611: xp fixture for array API tests (#619)

Saransh-cpp committed:
gh-402: add NumPy-like interface for JAX RNGs (#610)

Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-124: use `cosmology.api` over the `cosmology` package (#537)

Saransh-cpp committed:
gh-402: wrap NumPy RNGs to return array_api_strict Array (#622)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

Saransh-cpp committed:
gh-502: add array API tests and typing for `grf._transformations` (#614)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-621: bump `array-api-strict` version to `2024.12` (#625)

Saransh-cpp committed:
gh-519: test array API compliance of legacy mode functions (#595)

Saransh-cpp committed:
gh-594: split PyPI and TestPyPI deployments (#626)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#627)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
Disable PLC0415 in tests (#630)

paddyroddy committed:
gh-629: temporarily pin `array-api-strict` (#632)

dependabot[bot] committed:
DEV: Bump actions/download-artifact from 4 to 5 (#635)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
DEV: Bump actions/checkout from 4 to 5 (#636)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ntessore committed:
gh-639: fix compute_gaussian_spectra() for empty spectra (#640)

dependabot[bot] committed:
DEV: Bump actions/setup-python from 5 to 6 (#638)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

paddyroddy committed:
gh-654: add `types` for the labelled examples workflow (#655)

paddyroddy committed:
gh-652: make `array-api-strict` an official dependency (#653)

connoraird committed:
gh-407: Port simple functions in fields.py (#642)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#628)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Nicolas Tessore <n.tessore@ucl.ac.uk>

paddyroddy committed:
gh-633: switch to `dependency-groups` (#634)

paddyroddy committed:
gh-659: rename hook to `ruff-check` (#660)

connoraird committed:
gh-408: porting straightforward functions in `shells` (#643)

paddyroddy committed:
gh-669: Add test to make sure that GLASS is tested on its own (#670)

ntessore committed:
gh-673: all-contributors needs leading line (#674)

allcontributors[bot] committed:
docs: add connoraird as a contributor for code, review, and maintenance (#675)

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>

paddyroddy committed:
gh-667: Make JAX an optional dependency as intended (#672)

connoraird committed:
gh-409: Port straightforward functions in `points` (#663)

connoraird committed:
gh-676: Remove references to UnifiedGenerator (#681)

connoraird committed:
gh-679: only import numpy when it is absolutely necessary and add doc… (#680)

paddyroddy committed:
gh-682: create an `isort` section for Array API (#684)

paddyroddy committed:
gh-666: make docs fail on warnings (#685)

paddyroddy committed:
gh-687: fix documentation (#688)

paddyroddy committed:
gh-686: relax `array-api-strict` upper bound pin (#689)

paddyroddy committed:
gh-691: rename envvar to `ARRAY_BACKEND` (#692)

paddyroddy committed:
gh-693: Add script to generate release notes (#694)

paddyroddy committed:
gh-695: Prepare for the `2025.2` release (#696)

paddyroddy committed:
gh-700: remove `typing-extensions` dependency (#701)

connoraird committed:
gh-410: Port straightforward functions in `observations` (#677)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

ntessore committed:
gh-683: rewrite dispatch mechanism for GRF transformations (#697)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-702: use `ruff.lint.future-annotations` (#703)

Co-authored-by: Connor Aird <c.aird@ucl.ac.uk>

connoraird committed:
gh-704: Remove isinstance(x, Number) (#706)

paddyroddy committed:
gh-678: improved `mypy` test configuration via `pyproject.toml` (#710)

paddyroddy committed:
gh-623: use `array_api_compat.array_namespace` helper (#711)

ntessore committed:
gh-705: add displace() and displacement() (#708)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

connoraird committed:
gh-406: Add xp to tests for ported arraytools functions (#717)

paddyroddy committed:
gh-714: ensure `pytest` and `nox` agree on test dependencies (#715)

dependabot[bot] committed:
gh-725: Bump actions/download-artifact from 5 to 6 (#718)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
gh-725: Bump actions/upload-artifact from 4 to 5 (#719)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

paddyroddy committed:
gh-644: create a custom types file (#729)

connoraird committed:
gh-721: Port RNG functions in `shapes.py` (#722)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

connoraird committed:
gh-723: Port RNG functions in `galaxies.py` (#724)

Co-authored-by: Patrick J. Roddy <patrickjamesroddy@gmail.com>

connoraird committed:
gh-734: Port points.uniform_positions (#736)

paddyroddy committed:
gh-730: add missing import from docstring (#741)

paddyroddy committed:
gh-732: use the new array types from `glass.types` (#733)

paddyroddy committed:
gh-737: consistent importing of `_array_api_utils` (#738)

paddyroddy committed:
gh-747: Force nox to default to 3.13 (#748)

paddyroddy committed:
gh-727: ensure examples workflow runs on change (#728)

paddyroddy committed:
gh-749: Make `_uniform_grid` (and upstream) require array backend (#750)

paddyroddy committed:
gh-751: tidy up doc plotting routines (#752)

pre-commit-ci[bot] committed:
pre-commit.ci: update pre-commit hooks (#753)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

paddyroddy committed:
gh-739: don't use numpy alias when interacting with backend (#740)

paddyroddy committed:
gh-754: modern SPDX licensing (#755)

paddyroddy committed:
gh-720: add warning if `nearcorr` has no convergence (#757)

paddyroddy committed:
gh-762: general linting improvments (#763)

connoraird committed:
Add nox command to generate coverage report of glass

connoraird committed:
Generate coverage in github actions

connoraird committed:
Install coverage outside nox to allow xml generation in action

connoraird committed:
Simplify _get_revisions_from_posargs

paddyroddy commented on noxfile.py:
```diff
@@ -20,10 +20,40 @@
     "jax>=0.4.32",
     "pytest",
     "pytest-benchmark",
+    "coverage[toml]",
```
Sorry, I'm a pedant

```suggestion
"array_api_strict>=2",
"coverage[toml]",
"jax>=0.4.32",
"pytest",
"pytest-benchmark",
```

paddyroddy commented on noxfile.py:
```diff
@@ -20,10 +20,40 @@
+
+    if n == 1 and expected_num_revisions in {1, 2}:
+        return posargs[0], ""
+    if n == expected_num_revisions == 2:  # noqa: PLR2004
```
Would be much more readable

```suggestion
    if n in {expected_num_revisions, 2}:  # noqa: PLR2004
```

paddyroddy replied:
```suggestion
    if n == expected_num_revisions and n == 2:  # noqa: PLR2004
```

paddyroddy reviewed (changes requested):
Please right a description

connoraird commented on .github/workflows/test.yaml:
```diff
@@ -44,5 +44,29 @@ jobs:
+        run: nox -s coverage-${{ matrix.python-version }} -- main
+
+      - name: Install coverage.py outside of nox to generate xml
+        run: python -m pip install coverage[toml]
```
Possibly, I think I didn't incase the coverage versions installed by nox/pip got confused

connoraird committed:
Combine pip sections in action

connoraird committed:
Format noxfile and improve _get_revisions_from_posargs

connoraird committed:
Be more explicit with conditionals

connoraird committed:
gh-756: Add workflow to trigger reusable workflow from benchmarks repo (#765)

connoraird committed:
Attempt to display glass src in coveralls.io

connoraird committed:
Add coveragerc mapping glass pkg to glass/glass

paddyroddy commented on .coveragerc:
```diff
@@ -0,0 +1,10 @@
+[run]
```
Let's move this to `pyproject.toml`. See https://github.com/glass-dev/glass/blob/9bcdbe8e97c7548090bc5554cb7d2cc9300da30e/pyproject.toml#L88-L101

connoraird committed:
Swap to pyproject.toml in favour of .coveragerc

connoraird committed:
Dependencies use - instead of _

connoraird committed:
Add glass as a submodule to allow src to appear in coveralls

connoraird committed:
Remove _get_revisions_from_posargs as it no longer makes sense

connoraird committed:
Attempt to display glass src in coveralls.io tree

connoraird committed:
Add base-path to tell coveralls where to look for glass files

connoraird committed:
Revert "Add base-path to tell coveralls where to look for glass files"

This reverts commit 8534ea66615adb5bb7f914124256180434b89f64.

paddyroddy committed:
Test coveralls

paddyroddy committed:
Fix omit

paddyroddy committed:
Try `base-path`

paddyroddy committed:
Add slash

paddyroddy committed:
Change back to source

paddyroddy committed:
Don't install package

connoraird committed:
Change back to installing glass

paddyroddy committed:
Fix message

paddyroddy committed:
Upload coverage

paddyroddy committed:
Remove base path

paddyroddy committed:
Add relative files

paddyroddy committed:
gh-725: change @dependabot commit title (#726)

paddyroddy committed:
gh-742: make `allcontributors` follow the commit spec (#743)

paddyroddy committed:
Lint fix

paddyroddy committed:
Don't install the package

paddyroddy committed:
Try source dir

paddyroddy committed:
Exclude nox

paddyroddy committed:
Test coveralls

paddyroddy committed:
Try working dir

paddyroddy committed:
Undo workdir

paddyroddy committed:
Fix artefact

paddyroddy committed:
Adjust source

paddyroddy committed:
Remove source

paddyroddy committed:
Use `sed`

paddyroddy committed:
Change sed

paddyroddy committed:
Add `base-path`

paddyroddy committed:
Add debug

paddyroddy committed:
New attempt

paddyroddy committed:
Longer base path

paddyroddy committed:
Changes

paddyroddy committed:
Remove submodule

paddyroddy committed:
Add 'glass/' from commit 'b42d62f8c115b54e681c8a0219c2b1ce1b5f1574'

git-subtree-dir: glass
git-subtree-mainline: 4bcea98a7e0e5a4cfe4b225967e3c71ce90c9be6
git-subtree-split: b42d62f8c115b54e681c8a0219c2b1ce1b5f1574

paddyroddy committed:
Fix source

paddyroddy committed:
Back to pytst

paddyroddy committed:
Change to path

paddyroddy committed:
Remove artefact upload

paddyroddy committed:
Fix lint

paddyroddy commented:
Closing in favour of https://github.com/glass-dev/glass/pull/769
